common/src/main/kotlin/org/jetbrains/kotlinconf/DateFormat.kt

```diff
@@ -18,36 +18,40 @@ fun Pair<GMTDate, GMTDate>.toReadableString(): String = buildString {
     if (fromSuffix != toSuffix) {
         append("${from.toReadableTimeString()}$emDash{${to.toReadableTimeString()}}")
     } else {
-        append("${from.readableHours()}:${from.minutes.asMinutesString()}")
+        append("${from.readableHours()}:${from.minutes.toString(2)}")
         append(emDash)
-        append("${to.readableHours()}:${to.minutes.asMinutesString()} $fromSuffix")
+        append("${to.readableHours()}:${to.minutes.toString(2)} $fromSuffix")
     }
 }
 
 fun GMTDate.timeSuffix() = if ((hours + 11) / 12 == 1) "a.m." else "p.m."
 fun GMTDate.readableHours() = (hours + 11) % 12 + 1
 
-fun Int.asMinutesString(): String = if (this < 10) "0$this" else toString()
+fun Int.toString(minSize: Int): String = "$this".padStart(minSize, '0')
 
 private const val emDash = "\u2014"
 
-fun GMTDate.toReadableTimeString(): String = "${readableHours()}:${minutes.asMinutesString()} ${timeSuffix()}"
+fun GMTDate.toReadableTimeString(): String = "${readableHours()}:${minutes.toString(2)} ${timeSuffix()}"
 fun GMTDate.toReadableDateString(): String = "$year ${month.value} $dayOfMonth"
 fun GMTDate.toReadableDateTimeString() = "${toReadableDateString()} ${toReadableTimeString()}"
 
-fun GMTDate.Companion.renderDateInterval(from: GMTDate, to: GMTDate) = (from to to).toReadableString()
-
 /**
  * According to mask: "yyyy-MM-dd'T'HH:mm:ss"
  */
-fun parseDate(date: String): GMTDate = date.run {
-    val year = substring(0, 4).toInt()
-    val month = substring(5, 7).toInt()
-    val day = substring(8, 10).toInt()
+fun String.parseDate(): GMTDate {
+    fun formatError(): Nothing = throw Error("Format of $this is not correct")
```

Comment:
We could write a single catch for all `toInt()` and wrap exception instead.

Reply:
Done